zebra/zserv.c

```diff
@@ -1029,9 +1052,13 @@ void zserv_start(const char *path)
 
 void zserv_event(struct zserv *client, enum zserv_event event)
 {
+	struct accept_event *acc;
+
 	switch (event) {
 	case ZSERV_ACCEPT:
-		event_add_read(zrouter.master, zserv_accept, NULL, zsock, NULL);
```

Comment:
why can't we just test 'zsock' here and ... not try to use it if it's not valid? I guess it seems like there's the possibility that a stale event could be present, and we could a) find and cancel those events, or b) when the stale event runs, notice that it's stale, or c) add some new machinery. It seems like b) should be easier here?

Reply:
I think we can test if zsock is -1 in zserv_accept(). That way  zebra can return from zserv_accept() without reregistering itself and without accepting a new client request.